### ISSUE TITLE
Execution/sound on add task/comment/completion

### ISSUE THREAD

**Issue by alanvardy** [alanvardy/tod#786]
For people who want more than a terminal bell on success or failure we could get them to add a path to their config

`play_path_on_success` and `play_path_on_failure`. They will be nullable strings.

**Comment by stacksjb** [comment]
Thx! I will document example config for this. 

And down the road probably add something to config settings options so newbies don't have to manually edit config

**Comment by alanvardy** [comment]
Yes, I agree.

**Comment by stacksjb** [comment]
Also - does "success" and "failure" equal "task completion"? I was looking for "task completion" and "task creation", not specifically general command/code success/failure (though error/failure is good for terminal bell). 

This is basically to strictly replicate the dopamine fix when you check off the task in the official client 😬

**Comment by alanvardy** [comment]
If a task is completed or created that would be a `success` but yes, any successfully completed command is also a `success` and would trigger the bell.

Are you looking to play a sound file every time:
- You create a task
- You complete a task
- You complete each item in `list process` <- this too?

I'm sensitive to sounds and notifications, so this feature doesn't intuitively appeal to me, but I do like it getting a sound if something errors, and I need to look at it. Does the list above meet your needs? If yes, I can maybe do:

- `play_path_on_success`
- `play_path_on_failure`
- `play_path_on_completion`

**Comment by alanvardy** [comment]
Really good ideas. I can do the `*_on_completion` ones and will think about the execute command.

**Comment by alanvardy** [comment]
Rodio would probably be a good option for this:
https://github.com/RustAudio/rodio

**Comment by stacksjb** [comment]
Yes - I use Rodio to play sound which works great in the generalized wrapper I wrote. See https://github.com/stacksjb/Shell_Command_Menu/blob/main/src/utils.rs for an example of how I have it setup.

**Comment by stacksjb** [comment]
I'll add code for this. 

For my use case, success for the entire execution of the program doesn't meet what I need, so the bell_on_sucess/fail aren't right exactly what is needed, and I'd like it to be a bit more flexible rather than just written for a single specific use case.

**Comment by stacksjb** [comment]
@alanvardy from an architecture perspective, do you have a preference between
a) Calling a function every time, but the function acts according to config (so the function does nothing if not set in config)
or
b) Wrapping function in an if statement and only calling if config is set?

The second is more consistent and reusable, but the first is clearer in the code (I suppose there could be a third hybrid where you a call a function with the config is set but let's not get too complicated).

**Comment by alanvardy** [comment]
That's a good question. If the function is called in just a place or two I prefer `b` because it is more readable.

If it is used quite commonly (probably what you are doing here) I think I would prefer `a` and call it `maybe_` as in `maybe_ring_bell` to indicate that special functionality. I think there might be a few functions that need to be prefixed with maybe in the codebase already.

I come from the perspective that everything is a trade-off and patterns that work well in small codebases don't always translate well to a large codebase and vice versa. I guess I prioritize readability and maintainability over everything else, but that can be super subjective.

**Comment by alanvardy** [comment]
I think that this codebase is still relatively small but that the logic is actually quite dense. I've already forgotten a lot of what I have written and am already reading things that I wrote years ago as if someone else wrote it 😆

**Comment by stacksjb** [comment]
Initial PR Up - Looking for validation before I add the other two (since it will be mostly identical code)

**Comment by alanvardy** [comment]
Awesome, reviewed.
